cmd/up.go

```diff
@@ -366,6 +367,9 @@ func newUpCmd() *cobra.Command {
 	cmd.PersistentFlags().StringArrayVarP(
 		&configArray, "config", "c", []string{},
 		"Config to use during the update")
+	cmd.PersistentFlags().BoolVar(
+		&path, "path", false,
+		"Config keys contain a path to a property in a map or list to set")
```

Comment:
What does this flag do in practice? I might be missing something, but I don't see what this has to do with `pulumi up`. Right?

Reply:
`pulumi up` supports `-c` flags to augment config from the command line during an update. So does `pulumi preview` and `pulumi new`.

The `--path` flag is needed to opt-in to allowing paths for those. When Luke and I discussed the design, his feeling was that we should support paths everywhere we currently accept config keys as input to keep the CLI/experience consistent.

I'm open to other ways of opting-in to the path behavior (something other than using `--path`). But we do need _some_ way of opting-in, because periods and square brackets can be used in regular config key names today -- we can't just start interpreting those characters differently by default as that could break folks.

Reply:
BTW, I am open to using a different name rather than `--path`. The downside to path is that it might make folks think that it has something to do with file paths (like a path to a config file), which it is not.

Some other ideas:
 - `--key-path`
 - `--key-property-path`
 - `--property`
 - `--property-path`

For `pulumi [new|preview|up]`, it occurred to me that it might be worth prefixing the flag name with `config` so it is more clear that it is related to the keys passed via the `--config` (`-c`) flags. So if we keep the flag for `pulumi config` named `--path`, for `pulumi [new|preview|up]` it should be `--config-path`.

Reply:
The more I've been thinking about this the more I'm leaning towards using a name other than `--path` to opt-in to the property path behavior.

Additional ideas:
 - `--object`
 - `--prop`
 - `--prop-path`

Reply:
> I am open to using a different name rather than `--path`

We haven't found an alternative that's a clear net improvement over `--path`, so we'll keep it `--path`.

> For `pulumi [new|preview|up]`, it occurred to me that it might be worth prefixing the flag name with `config` so it is more clear that it is related to the keys passed via the `--config` (`-c`) flags.

I did go ahead and prefix the flag as `--config-path` for `pulumi [new|preview|up]`.